src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs

```diff
@@ -194,38 +194,52 @@ internal static void SetLogRecordSeverityFields(ref LogRecordData logRecordData,
             iLoggerData.State = state;
             return null;
         }
+#if NET5_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
+        else if (RuntimeFeature.IsDynamicCodeSupported)
+        {
+            return ParseCustomState(logRecord, state);
+        }
+#endif
         else
         {
-            try
-            {
-                PropertyDescriptorCollection itemProperties = TypeDescriptor.GetProperties(state);
+            OpenTelemetrySdkEventSource.Log.LoggerProcessStateSkipped<TState>();
```

Comment:
Why skip processing the state for older targets?

Reply:
RuntimeFeature.IsDynamicCodeSupported only applies to these conditions: <html>
<body>
<!--StartFragment-->

.NET | Core 3.0, Core 3.1, 5, 6, 7, 8
-- | --

<!--EndFragment-->
</body>
</html> https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimefeature.isdynamiccodesupported?view=net-8.0

@vitek-karas, @eerhardt, would `.IsDynamicCodeSupported` always evaluates to false when the publishing AOT flag is set to true?

Reply:
>would .IsDynamicCodeSupported always evaluates to false when the publishing AOT flag is set to true?

Correct. When an app has `PublishAot=true` in its .csproj, `IsDynamicCodeSupported` will report `false`.